### PR TITLE
Corrections to issues raised by Cecil Phillip

### PR DESCRIPTION
**Pull request by robvet**
## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

File: docs/architecture/cloud-native/combine-containers-serverless-approaches.md [hunk 1]
```diff
@@ -28,9 +28,17 @@ func init ProjectName --worker-runtime dotnet --docker
 
 When the project is created, it will include a Dockerfile and the worker runtime configured to `dotnet`. Now, you can create and test your function locally. Build and run it using the  `docker build` and `docker run` commands. For detailed steps to get started building Azure Functions with Docker support, see the [Create a function on Linux using a custom image](https://docs.microsoft.com/azure/azure-functions/functions-create-function-linux-custom-image) tutorial.
 
+When creating Azure functions, you'll want to reference version 3 of the Azure Functions runtime, which targets .NET Core 3.1 and later. [This link](https://docs.microsoft.com/azure/azure-functions/functions-versions) describes the runtime versions available for Azure Functions.
```
Comment (nishanil) [inline docs/architecture/cloud-native/combine-containers-serverless-approaches.md:31]: Remove duplicate content.

File: docs/architecture/cloud-native/candidate-apps.md [hunk 2]
```diff
@@ -68,7 +68,7 @@ With the introduction behind, we now dive into a much more detailed look at clou
 
 - [Cloud Native Computing Foundation](https://www.cncf.io/)
 
-- [.NET Microservices: Architecture for Containerized .NET applications](https://dotnet.microsoft.com/download/thank-you/microservices-architecture-ebook)
+- [.NET Microservices: Architecture for Containerized .NET applications](https://docs.microsoft.com/dotnet/standard/microservices-architecture/)
```
Comment (nishanil) [inline docs/architecture/cloud-native/candidate-apps.md:71]: earlier link is correct and helps in tracking downloads. Can you revert these changes?

File: docs/architecture/cloud-native/deploy-containers-azure.md [hunk 3]
```diff
@@ -103,8 +103,9 @@ Developers share a running (development) instance in an AKS cluster that contain
...
 
-> Imagine you're a new employee trying to fix a bug in a complex microservices application consisting of dozens of components, each with their own configuration and backing services. To get started, you must configure your local development environment so that it can mimic production including setting up your IDE, building tool chain, containerized service dependencies, a local Kubernetes environment, mocks for backing services, and more. With all the time involved setting up your development environment, fixing that first bug could take days.
-> Or you could use Dev Spaces and AKS.
+"Imagine you're a new employee trying to fix a bug in a complex microservices application consisting of dozens of components, each with their own configuration and backing services. To get started, you must configure your local development environment so that it can mimic production including setting up your IDE, building tool chain, containerized service dependencies, a local Kubernetes environment, mocks for backing services, and more. With all the time involved setting up your development environment, fixing that first bug could take days.
```
Comment (nishanil) [inline docs/architecture/cloud-native/deploy-containers-azure.md:106]: Drop these changes. Merge issue here.

File: docs/architecture/cloud-native/deploy-eshoponcontainers-azure.md [hunk 4]
```diff
@@ -1,39 +1,48 @@
...
+You can now deploy the eShop application to the cluster leveraging Helm and Tiller.
 
 ## Deploying to Azure Kubernetes Service using Helm
 
-Basic deployments to AKS may use custom CLI scripts or simple deployment files, but more complex applications should use a dependency management tool like Helm. Helm is maintained by the Cloud-native Computing Foundation and helps you define, install, and upgrade Kubernetes applications. Helm is composed of a command-line client, helm, which uses helm charts, and an in-cluster component, Tiller. Helm Charts use standard YAML-formatted files to describe a related set of Kubernetes resources and are typically versioned alongside the application they describe. Helm Charts range from simple to complex depending on the requirements of the installation they describe.
+Helm is an application package manager tool that works directly with Kubernetes. It helps you define, install, and upgrade Kubernetes applications. While simple apps can be deployed to AKS with custom CLI scripts or simple deployment files, complex apps can contain many Kubernetes objects and benefit from Helm.
```
Comment (nishanil) [inline docs/architecture/cloud-native/deploy-eshoponcontainers-azure.md:21]: Can you provide an example of a Helm Chart?

File: docs/architecture/cloud-native/front-end-communication.md [hunk 5]
```diff
@@ -119,7 +119,7 @@ Azure API Management is available across [four different tiers](https://azure.mi
...
 
-Recently, Microsoft announced a [API Management serverless tier](https://azure.microsoft.com/blog/announcing-azure-api-management-for-serverless-architectures/) for Azure API Management. Referred to as the *consumption pricing tier*, the service is a variant of API Management designed around the serverless computing model. Unlike the "pre-allocated" pricing tiers previously shown, the consumption tier provides  instant provisioning and pay-per-action pricing.
+Recently, Microsoft released a [serverless tier](https://azure.microsoft.com/blog/announcing-azure-api-management-for-serverless-architectures/) for Azure API Management. Referred to as the *consumption pricing tier*, the service is a variant of API Management designed around the serverless computing model. Unlike the "pre-allocated" pricing tiers previously shown, the consumption tier provides  instant provisioning and pay-per-action pricing.
```
Comment (nishanil) [inline docs/architecture/cloud-native/front-end-communication.md:122]: Not recently. Consider rewriting..

File: docs/architecture/cloud-native/grpc.md [hunk 6]
```diff
@@ -48,7 +48,7 @@ The book, [gRPC for WCF Developers](https://docs.microsoft.com/dotnet/architectu
 
 ## gRPC support in .NET
 
-gRPC is integrated into .NET Core 3.0 SDK or later. The following tools support it:
+gRPC is integrated into .NET Core 3.1 SDK or later. The following tools support it:
```
Comment (nishanil) [inline docs/architecture/cloud-native/grpc.md:51]: It should be 3.0 and later?

File: docs/architecture/cloud-native/grpc.md [hunk 7]
```diff
@@ -98,7 +98,7 @@ In the previous figure, note how eShop embraces the [Backend for Frontends patte
 
 **Figure 4-23**. gRPC in eShop on Containers
 
-gRPC communication requires both client and server components. In the previous figure, note how the Shopping Aggregator implements a gRPC client. The client makes synchronous gRPC calls (in red) to backend microservices, each of which implement a gRPC server. Both the client and server take advantage of the built-in gRPC plumbing from the .NET Core 3.0 SDK. Client-side *stubs* provide the plumbing to invoke remote gRPC calls. Server-side components provide gRPC plumbing that custom service classes can inherit and consume.
+gRPC communication requires both client and server components. In the previous figure, note how the Shopping Aggregator implements a gRPC client. The client makes synchronous gRPC calls (in red) to backend microservices, each of which implement a gRPC server. Both the client and server take advantage of the built-in gRPC plumbing from the .NET Core 3.1 SDK. Client-side *stubs* provide the plumbing to invoke remote gRPC calls. Server-side components provide gRPC plumbing that custom service classes can inherit and consume.
```
Comment (nishanil) [inline docs/architecture/cloud-native/grpc.md:101]:
```suggestion
gRPC communication requires both client and server components. In the previous figure, note how the Shopping Aggregator implements a gRPC client. The client makes synchronous gRPC calls (in red) to backend microservices, each of which implements a gRPC server. Both the client and server take advantage of the built-in gRPC plumbing from the .NET Core SDK. Client-side *stubs* provide the plumbing to invoke remote gRPC calls. Server-side components provide gRPC plumbing that custom service classes can inherit and consume.
```

File: docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md [hunk 8]
```diff
@@ -64,7 +64,7 @@ The code is organized to support the different microservices, and within each mi
 
 ## Understanding microservices
 
-This book focuses on cloud-native applications built using Azure technology. To learn more about microservices best practices and how to architect microservice-based applications, read the companion book, [.NET Microservices: Architecture for Containerized .NET Applications](https://dotnet.microsoft.com/learn/aspnet/microservices-architecture). The book is available online, in PDF, or eReader formats.
+This book focuses on cloud-native applications built using Azure technology. To learn more about microservices best practices and how to architect microservice-based applications, read the companion book, [.NET Microservices: Architecture for Containerized .NET Applications](https://docs.microsoft.com/dotnet/standard/microservices-architecture/). The book is available online, in PDF, or eReader formats.
```
Comment (nishanil) [inline docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md:67]: Change link to the PDF link referenced earlier.

File: docs/architecture/cloud-native/key-takeaways.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
+---
+title: Key takeaways
+description: Get the key takeaways from the Architecting Cloud-Native .NET Apps for Azure guide/e-book.
+ms.date: 04/28/2020
+---
+
+# Key Takeaways
```
Comment (nishanil) [inline docs/architecture/cloud-native/key-takeaways.md:7]: Can we change this to Summary?

File: docs/architecture/cloud-native/service-mesh-communication-infrastructure.md [hunk 10]
```diff
@@ -49,7 +49,7 @@ We next move to distributed data in cloud-native systems and the benefits and ch
 
 ### References
 
-- [.NET Microservices: Architecture for Containerized .NET applications](https://dotnet.microsoft.com/download/thank-you/microservices-architecture-ebook)
+- [.NET Microservices: Architecture for Containerized .NET applications](https://docs.microsoft.com/dotnet/standard/microservices-architecture/)
```
Comment (nishanil) [inline docs/architecture/cloud-native/service-mesh-communication-infrastructure.md:52]: The earlier link is fine.

**Review by nishanil — COMMENTED** [review]
I left some comments for you to review.

File: docs/architecture/cloud-native/combine-containers-serverless-approaches.md (hunk 1 above)
Comment (robvet) [reply docs/architecture/cloud-native/combine-containers-serverless-approaches.md:31]: So, think you suggesting removing the item in green. It's definitely misplaced. This paragraph should not be under the heading "Combine serverless and docker containers." Would it make sense to move this somewhere else in the chapter, as it talks runtime versions for Functions?  Let me know what you think.

File: docs/architecture/cloud-native/candidate-apps.md (hunk 2 above)
Comment (robvet) [reply docs/architecture/cloud-native/candidate-apps.md:71]: Done

File: docs/architecture/cloud-native/deploy-containers-azure.md (hunk 3 above)
Comment (robvet) [reply docs/architecture/cloud-native/deploy-containers-azure.md:106]: Reverted back making the quote a 'blockquote'

File: docs/architecture/cloud-native/deploy-eshoponcontainers-azure.md (hunk 4 above)
Comment (robvet) [reply docs/architecture/cloud-native/deploy-eshoponcontainers-azure.md:21]: Done

File: docs/architecture/cloud-native/front-end-communication.md (hunk 5 above)
Comment (robvet) [reply docs/architecture/cloud-native/front-end-communication.md:122]: Done

File: docs/architecture/cloud-native/grpc.md (hunk 7 above)
Comment (robvet) [reply docs/architecture/cloud-native/grpc.md:101]: Removed version number.

File: docs/architecture/cloud-native/key-takeaways.md (hunk 9 above)
Comment (robvet) [reply docs/architecture/cloud-native/key-takeaways.md:7]: Done

File: docs/architecture/cloud-native/service-mesh-communication-infrastructure.md (hunk 10 above)
Comment (robvet) [reply docs/architecture/cloud-native/service-mesh-communication-infrastructure.md:52]: Reverted

File: docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md (hunk 8 above)
Comment (robvet) [reply docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md:67]: Reverted

File: docs/architecture/cloud-native/grpc.md (hunk 6 above)
Comment (robvet) [reply docs/architecture/cloud-native/grpc.md:51]: Changed

File: docs/architecture/cloud-native/combine-containers-serverless-approaches.md (hunk 1 above)
Comment (robvet) [reply docs/architecture/cloud-native/combine-containers-serverless-approaches.md:31]: Never mind - removed the paragraph.

File: docs/architecture/cloud-native/cloud-native-data.md [hunk 11]
```diff
@@ -0,0 +1,149 @@
+---
```
Comment (nishanil) [inline docs/architecture/cloud-native/cloud-native-data.md:1]: Drop this file.

Comment (robvet) [reply docs/architecture/cloud-native/cloud-native-data.md:1]: Removed. I see what I did. I merged updated master into my branch (duh). Should have merged my branch  into updated master. What I get for working 12 hours a day! :)

File: docs/architecture/cloud-native/combine-containers-serverless-approaches.md [hunk 12]
```diff
@@ -28,9 +28,15 @@ func init ProjectName --worker-runtime dotnet --docker
 
 When the project is created, it will include a Dockerfile and the worker runtime configured to `dotnet`. Now, you can create and test your function locally. Build and run it using the  `docker build` and `docker run` commands. For detailed steps to get started building Azure Functions with Docker support, see the [Create a function on Linux using a custom image](https://docs.microsoft.com/azure/azure-functions/functions-create-function-linux-custom-image) tutorial.
 
+For detailed steps to get started building Azure Functions with Docker support, see the [Create a function on Linux using a custom image](https://docs.microsoft.com/azure/azure-functions/functions-create-function-linux-custom-image) tutorial.
```
Comment (nishanil) [inline docs/architecture/cloud-native/combine-containers-serverless-approaches.md:31]: This is a merge issue.  The statement from the previous paragraph is repeated.

File: docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md [hunk 13]
```diff
@@ -64,7 +64,7 @@ The code is organized to support the different microservices, and within each mi
 
 ## Understanding microservices
 
-This book focuses on cloud-native applications built using Azure technology. To learn more about microservices best practices and how to architect microservice-based applications, read the companion book, [.NET Microservices: Architecture for Containerized .NET Applications](https://dotnet.microsoft.com/learn/aspnet/microservices-architecture). The book is available online, in PDF, or eReader formats.
+This book focuses on cloud-native applications built using Azure technology. To learn more about microservices best practices and how to architect microservice-based applications, read the companion book, [.NET Microservices: Architecture for Containerized .NET Applications](https://dotnet.microsoft.com/download/thank-you/microservices-architecture-ebook). The book is available online, in PDF, or eReader formats.
```
Comment (nishanil) [inline docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md:67]:
```suggestion
This book focuses on cloud-native applications built using Azure technology. To learn more about microservices best practices and how to architect microservice-based applications, read the companion book, [.NET Microservices: Architecture for Containerized .NET Applications](https://dotnet.microsoft.com/download/thank-you/microservices-architecture-ebook).
```

Comment (nishanil) [reply docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md:67]: Since we do not support e-reader formats and the link is directly targeting the pdf, we don't need to have this sentence.

File: docs/architecture/cloud-native/combine-containers-serverless-approaches.md (hunk 12 above)
Comment (robvet) [reply docs/architecture/cloud-native/combine-containers-serverless-approaches.md:31]: Good catch. Corrected

File: docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md (hunk 13 above)
Comment (robvet) [reply docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md:67]: Fixed

**Review by nishanil — APPROVED** [review]
